src/Nethermind/Nethermind.Evm/Instructions/EvmInstructions.Environment.cs

```diff
@@ -87,17 +87,17 @@ public interface IOpEnvUInt64
     /// <param name="programCounter">The program counter.</param>
     /// <returns>An EVM exception type if an error occurs.</returns>
     [SkipLocalsInit]
-    public static EvmExceptionType InstructionEnvBytes<TOpEnv>(VirtualMachine vm, ref EvmStack stack, ref long gasAvailable, ref int programCounter)
-        where TOpEnv : struct, IOpEnvBytes
+    public static EvmExceptionType InstructionEnvAddress<TOpEnv>(VirtualMachine vm, ref EvmStack stack, ref long gasAvailable, ref int programCounter)
+        where TOpEnv : struct, IOpEnvAddress
     {
         // Deduct the gas cost as defined by the operation implementation.
         gasAvailable -= TOpEnv.GasCost;
 
         // Execute the operation and retrieve the result.
-        TOpEnv.Operation(vm.EvmState, out Span<byte> result);
+        Address result = TOpEnv.Operation(vm.EvmState);
```

Comment:
should we have address struct?

Reply:
Doesn't really help as the array backing is used everywhere; would be a large change to alter it